src/adapter/src/explain/optimizer_trace.rs

```diff
@@ -72,7 +71,8 @@ impl OptimizerTrace {
                 .with(PlanTrace::<DataflowDescription<OptimizedMirRelationExpr>>::new(filter))
                 .with(PlanTrace::<DataflowDescription<Plan>>::new(filter))
                 // Don't filter for FastPathPlan entries (there can be at most one).
-                .with(PlanTrace::<FastPathPlan>::new(None));
+                .with(PlanTrace::<FastPathPlan>::new(None))
+                .with(PlanTrace::<UsedIndexes>::new(None));
```

Comment:
This adds a new `tracing_subscriber::Layer` which facilitates the recording of `UsedIndexes` structs with `trace_plan`.